utility_functions/date_period.py

```diff
@@ -63,6 +64,40 @@ def date_period(period: int, start_date: str = ''):
     return start_date, end_date
 
 
+def add_ltm_period(ih, date_col_name='invc_dt', date_format='YYYYMMDD'):
+    """
```

Comment:
Might want to use type hints for the inputs:

`
from pyspark.sql.dataframe import DataFrame as sparkDataFrame
.....
def add_ltm_period(
ih : sparkDataFrame,
date_col_name: str = 'invc_dt',
date_format: str = 'YYYYMMDD',
) ->sparkDataFrame:

`

Reply:
Added the type hints, though I didn't check your comment completely so I used DataFrame instead of sparkDataFrame - idea is the same for both, though I can change if you'd rather have it as sparkDataFrame

Reply:
I would think we might want to separate sparkDataFrame and pandasDataFrame since I can see DataFrame might be confusing to people who could be using both.    And in this case, the function only works on sparkDataFrame.

Reply:
Sure, no problem - just changed the type hint to sparkDataFrame